src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_EditProfileScreen.tsx

```diff
@@ -0,0 +1,324 @@
+    setSubmissionStatus('loading');
+
+    if (isValidEmail(email)) {
+      const {available} = await emailAvailable(email);
+      if (available || customerProfile?.email === email) {
+        try {
+          await updateProfile({
+            firstName,
+            surname,
+            email,
+          });
+          await fetchProfile();
+          setSubmissionStatus('submitted');
+        } catch {
+          setSubmissionStatus('submission-error');
+        }
+      } else {
+        setSubmissionStatus('unavailable-email');
+      }
+    } else {
+      setSubmissionStatus('invalid-email');
+    }
+  };
+
+  const getEmailErrorText = (
+    status: SubmissionStatus | undefined,
+  ): string | undefined => {
+    switch (status) {
+      case 'invalid-email':
+        return t(EditProfileTexts.personalDetails.email.formattingError);
+      case 'unavailable-email':
+        return t(EditProfileTexts.personalDetails.email.unavailableError);
+    }
+  };
+
+  const prefillPersonalia = (
+    response: string,
+    set: React.Dispatch<React.SetStateAction<string>>,
+  ) => {
+    // Receiving "_" from Entur when firstName or surname are not set on user profile, instead setting them to ""
+    if (response === '_') {
+      set('');
+    } else if (response) {
+      set(response);
+    }
+  };
+
+  const fetchProfile = useCallback(async () => {
+    try {
+      const response = await getProfile();
+      setCustomerProfile(response);
+
+      prefillPersonalia(response.email, setEmail);
+      prefillPersonalia(response.firstName, setFirstName);
+      prefillPersonalia(response.surname, setSurname);
+
+      setProfileState('success');
+    } catch {
+      setProfileState('error');
+    }
+  }, []);
+
+  useEffect(() => {
+    const getProfileData = async () => {
+      setProfileState('loading');
+      await fetchProfile();
+    };
+    getProfileData();
+  }, [fetchProfile]);
+
+  return (
+    <FullScreenView
+      headerProps={{
+        title: t(EditProfileTexts.header.title),
+        leftButton: {type: 'back', withIcon: true},
+      }}
+      parallaxContent={() => (
+        <View style={styles.parallaxContent}>
+          <ThemeText
+            type="heading--medium"
+            color="background_accent_0"
+            style={{flexShrink: 1}}
+          >
+            {t(EditProfileTexts.header.title)}
+          </ThemeText>
+        </View>
+      )}
+    >
+      {authenticationType !== 'phone' ? (
+        <View style={styles.noAccount}>
+          <ThemeText>{t(EditProfileTexts.notProfile)}</ThemeText>
+        </View>
+      ) : (
+        <>
+          <View style={styles.personalDetails}>
+            {profileState === 'loading' && <ActivityIndicator size="large" />}
+            <ThemeText accessibilityRole="header" color="secondary">
+              {t(EditProfileTexts.personalDetails.header)}
+            </ThemeText>
+          </View>
+          {profileState === 'error' ? (
+            <MessageBox
```

Comment:
This box should probably also be wrapped with a `<Section withPadding>` I think

Reply:
`Section` should only be used around section items (`ButtonSectionItem`, `LinkSectionItem`) etc.

Reply:
Right. Maybe just wrap it with a `View` then, to get that horizontal padding